### ISSUE TITLE
Error when displaying samples on LED Matrix

### ISSUE THREAD

**Issue by kjb05** [hzeller/rpi-rgb-led-matrix#574]
Hi, I've tried a couple different samples but every time I run them i get this output on my board:
![img_3911](https://user-images.githubusercontent.com/8227166/37854297-a50cee68-2ec0-11e8-9576-92bd199efff9.JPG)

Any ideas?

Thanks in advance for any help!

**Comment by angelogoncalve** [comment]
Try do this:

from rgbmatrix import RGBMatrix, RGBMatrixOptions
import time

number_of_rows = 32
number_of_panels = 2
parallel = 1
number_of_columns = 32

options = RGBMatrixOptions()


def rgbmatrix_options():
    #options = RGBMatrixOptions()
    options.multiplexing = 1
    options.row_address_type = 0
    options.brightness = 100
    options.rows = number_of_rows / 1
    options.cols = number_of_columns
    options.chain_length = 1 * number_of_panels
    options.parallel = parallel
    options.hardware_mapping = 'regular'
    options.inverse_colors = False
    options.led_rgb_sequence = "RGB"
    options.gpio_slowdown = 4
    options.pwm_lsb_nanoseconds = 130
    options.show_refresh_rate = 0
    options.disable_hardware_pulsing = True
    options.scan_mode = 1
    options.pwm_bits = 1
    options.daemon = 0
    options.drop_privileges = 0
    options.pixel_mapper_config = "Rotate:0"
    
    return RGBMatrix(options=options)


display = rgbmatrix_options()


height = display.height
width = display.width


for x in range(0, width):
    for y in range(0, height):
        
        display.SetPixel(x, y, 0, 255, 0)
        
        point = Point(x, y)
        print point
        
        time.sleep(1)

**Comment by hzeller** [comment]
Check that address-line `C` is properly connected. It seems to be clamped to ground.

**Comment by kjb05** [comment]
Sorry I am totally new to this, what do you mean by address line c?

**Comment by hzeller** [comment]
The line `C` as described in the wiring diagram:
  https://github.com/hzeller/rpi-rgb-led-matrix/blob/master/wiring.md#wiring-diagram
If you are using an adapter, check if everything from GPIO pin 18 to the chip and from there to the display (PCB and wiring) is working.

**Comment by kjb05** [comment]
I dont have any tools to actually test this the correct way but I checked to make sure everything was soldered correctly and it is. I fixed a pin that might have had a poor connection and when i tried to run another sample i got this
![img_3916](https://user-images.githubusercontent.com/8227166/37859069-8cfbd27a-2ee4-11e8-831e-4ed4ab7b2a05.jpg)

The same thing just with different colors. Can you think of anything else that may be causing this besides a bad GPIO connection?

**Comment by hzeller** [comment]
What are the current commandline flags you give to the programs ?
Can you take a picture of the output of
```
 ./demo --led-cols=64 --led-rows=32 -D3
```
?

**Comment by kjb05** [comment]
Is this what you mean?
![img_3918](https://user-images.githubusercontent.com/8227166/37859434-0aaedfc8-2eea-11e8-8014-ccf6e3fb66d3.jpg)

**Comment by hzeller** [comment]
No, in the `examples-api-use/` https://github.com/hzeller/rpi-rgb-led-matrix/tree/master/examples-api-use

**Comment by kjb05** [comment]
Ok, here is what i receive when running that commmandline:
![image1](https://user-images.githubusercontent.com/8227166/37861250-373836f8-2f0c-11e8-8a3f-8eb514942c6d.jpeg)
![image2](https://user-images.githubusercontent.com/8227166/37861252-394b2fea-2f0c-11e8-8e7d-1b04bfaec816.jpeg)

**Comment by hzeller** [comment]
It's uppercase `-D`

**Comment by kjb05** [comment]
Ah, well here is the output for that 
![image1 1](https://user-images.githubusercontent.com/8227166/37866585-0eb37cae-2f63-11e8-9e13-171e080d2b4a.jpeg)

Nothing was displayed on the LED Board

**Comment by hzeller** [comment]
you have compiled the library with adafruit hat. Is this the adapter you are using?

**Comment by kjb05** [comment]
Yes - this product https://www.adafruit.com/product/2345

**Comment by hzeller** [comment]
ok, so starting from here, you need to make it work,  e.g. check power, make sure to have the right address mode and multiplexing etc. The readmed help with all the details. Make sure to use --Led-rows _this_ library and read this readme.

**Comment by marcmerlin** [comment]
With apologies to everyone, this project is doing a one time cleanup of 
all the bugs that have not been addressed and closed in the last 10 years.

If you filed an issue to ask for help, please do not re-open this one or a new
bug, but go here instead: https://rpi-rgb-led-matrix.discourse.group/

Please be mindful that this project does not have any full time maintainers
or anyone paid. You should not expect personal help for your own projects or
answers to programming questions but you can ask for help with panels that
aren't working, setup questions, or anything that may be a bug here:
https://rpi-rgb-led-matrix.discourse.group/
If you need programming help and are willing to pay someone for it, please
state so in your message, and someone may be interested.

If you did report an actual bug that is still not fixed today, please accept
our apologies for this being mass closed, you can absolutely send a followup
and re-open the issue, or open a new up to date one as applicable.
Note that we very much welcome help, so if you can help fix your bug, you will
be a superstar :)

If your bug was linked to a pull request that was not resolved, please update
the pull request and the issue, and we will do our best to review them. We do
thank you for your undertanding that we have very limited free time, so any
extra work you do along with your bug report or PR request will be very much
appreciated and improve the odds that they will be looked at.

If you want to help contribute as a python or C# or any other language bindings,
you are also a superstar, please state so in the issue and we'll do our best to
give you the access and reviews you need.

For anything related to PWM panel support, please make sure you read README and
then refer to https://github.com/hzeller/rpi-rgb-led-matrix/issues/466 . 
Please do not comment if you're haven't read everything and/or if you're just
asking when your panel is going to be supported. Those bugs are if you're 
able to contribute to the effort.

For rPI5: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1603

For anything python, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
For anything C#, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
